vispy/scene/visuals.py

```diff
@@ -103,7 +104,9 @@ def create_visual_node(subclass):
 
     # Decide on new class name
     clsname = subclass.__name__
-    assert clsname.endswith('Visual')
+    if not (clsname.endswith('Visual') and issubclass(subclass, Frozen)):
+        raise RuntimeError('Class "%s" must end with Visual, and must '
```

Comment:
Why not just have VisualNode subclass from Frozen? That covers most of the user-facing API without making any extra demands on the visuals.